subsys/net/l2/ethernet/arp.c

```diff
@@ -179,14 +179,14 @@ static void arp_entry_register_pending(struct arp_entry *entry)
 
 static void arp_request_timeout(struct k_work *work)
 {
-	s64_t current = k_uptime_get();
+	u32_t current = k_uptime_get_32();
 	struct arp_entry *entry, *next;
 
 	ARG_UNUSED(work);
 
 	SYS_SLIST_FOR_EACH_CONTAINER_SAFE(&arp_pending_entries,
 					  entry, next, node) {
-		if ((entry->req_start + ARP_REQUEST_TIMEOUT - current) > 0) {
+		if (entry->req_start + ARP_REQUEST_TIMEOUT > current) {
```

Comment:
Will this work properly if req_start + ARP_REQ_TIMEOUT wraps over?
We could do it in similar way as how `include/net/net_ip.h:net_tcp_seq_greater()` does it.